src/main/scala/org/apache/spark/sql/BebeFunctions.scala

```diff
@@ -39,6 +39,23 @@ object BebeFunctions {
       BeginningOfMonth(col.expr)
     }
 
+  def endOfDay(col: Column, timeZoneId: Option[String] = None): Column =
+    withExpr {
+      val startOfNextDay =
+        TruncTimestamp(lit("day").expr, DateAdd(col.expr, lit(1).expr), timeZoneId)
+      // TODO: Is this definition of end of day common? Maybe we shouldn't have an
+      // endOfDay method at all.
+      TimeAdd(startOfNextDay, expr("interval -1 microsecond").expr, timeZoneId)
+    }
+
+  def endOfMonth(col: Column): Column =
+    // TODO: I didn't follow why we need BeginningOfMonth, so I didn't add an EndOfMonth
+    // expression.
```

Comment:
I tried to avoid adding a `BeginningOfMonth` expression, [see the conversation here](https://github.com/MrPowers/bebe/pull/4), but couldn't figure out any other way to do it.  We should avoid adding expressions whenever possible - leverage what's built into Spark whenever we can.

Reply:
Couldn't `beginningOfMonth` be written like I've done `endOfMonth`, i.e. without a dedicated expression?

Reply:
@nchammas - Yea, maybe `beginningOfMonth` could have been written without a dedicated expression.  I couldn't figure it out, but if you see a better way, please send a PR and show me the light!!